library/xml-model/src/main/java/org/cru/godtools/xml/model/Manifest.kt

```diff
@@ -205,88 +220,85 @@ class Manifest : BaseModel, Styles {
+        require(XmlPullParser.START_TAG, XMLNS_MANIFEST, XML_PAGES)
 
         // process any child elements
-        val result = PagesData()
-        result.tractPages = buildList {
-            while (parser.next() != XmlPullParser.END_TAG) {
-                if (parser.eventType != XmlPullParser.START_TAG) continue
-
-                when (parser.namespace) {
-                    XMLNS_MANIFEST -> when (parser.name) {
-                        XML_PAGES_PAGE -> {
-                            val fileName = parser.getAttributeValue(null, XML_PAGES_PAGE_FILENAME)
-                            val srcFile = parser.getAttributeValue(null, XML_PAGES_PAGE_SRC)
-                            val pos = size
-                            parser.skipTag()
-
-                            if (srcFile != null)
-                                add(async { parseFile(srcFile).use { TractPage(this@Manifest, pos, fileName, it) } })
+        while (next() != XmlPullParser.END_TAG) {
+            if (eventType != XmlPullParser.START_TAG) continue
+
+            when (namespace) {
+                XMLNS_MANIFEST -> when (name) {
+                    XML_PAGES_PAGE -> {
+                        val fileName = getAttributeValue(null, XML_PAGES_PAGE_FILENAME)
+                        val src = getAttributeValue(null, XML_PAGES_PAGE_SRC)
+                        skipTag()
+
+                        if (src != null) {
+                            val pos = result.tractPageTasks.size
+                            result.tractPageTasks += scope.async {
+                                parseFile(src).use { TractPage(this@Manifest, pos, fileName, it) }
+                            }
                         }
-                        else -> parser.skipTag()
                     }
-                    XMLNS_ARTICLE -> when (parser.name) {
-                        XML_PAGES_AEM_IMPORT -> {
-                            parser.getAttributeValueAsUriOrNull(XML_PAGES_AEM_IMPORT_SRC)
-                                ?.let { result.aemImports.add(it) }
-                            parser.skipTag()
-                        }
-                        else -> parser.skipTag()
+                    else -> skipTag()
+                }
+                XMLNS_ARTICLE -> when (name) {
+                    XML_PAGES_AEM_IMPORT -> {
+                        getAttributeValueAsUriOrNull(XML_PAGES_AEM_IMPORT_SRC)?.let { result.aemImports += it }
+                        skipTag()
                     }
-                    else -> parser.skipTag()
+                    else -> skipTag()
                 }
+                else -> skipTag()
             }
-        }.awaitAll()
-        return@runBlocking result
+        }
     }
 
     @WorkerThread
-    private fun parseResources(parser: XmlPullParser): Map<String?, Resource> {
-        parser.require(XmlPullParser.START_TAG, XMLNS_MANIFEST, XML_RESOURCES)
+    private fun XmlPullParser.parseResources() = buildList {
```

Comment:
I defined the `XmlPullParser` as the "receiver" for the method. This means that within the method the "receiver" is referenced via `this`.

See: https://kotlinlang.org/docs/reference/extensions.html

Reply:
Ah, I see because you prefixed it with ```XmlPullParser``` that extends the function to be able to use methods from the ```XmlPullParser``` type. So anywhere you have ```this@``` the ```this``` is referring to the XmlPullParser and then calling a method of that like ```this@parseResources```.

Reply:
the `@{name}` syntax on `this` helps disambiguate which `this` you are referring to.

By overloading `this` with receivers and inline lambdas you can easily hit a point where multiple objects can be referenced by `this` and there can be overlap with the method & property names between those objects. the `@{name}` syntax indicates explicitly which `this` you are referring to

Reply:
https://kotlinlang.org/docs/reference/this-expressions.html